test/src/rabbit_hash_exchange_test.erl

```diff
@@ -62,10 +62,14 @@ test0(MakeMethod, MakeMsg, Queues, MsgCount, Count) ->
                                                  routing_key = <<"">> })
      || Q <- Queues],
 
+    amqp_channel:call(Chan, #'confirm.select'{}),
+
     [amqp_channel:call(Chan,
                        MakeMethod(),
                        MakeMsg()) || _ <- lists:duplicate(MsgCount, const)],
 
+    amqp_channel:wait_for_confirms_or_die(Chan),
+
```

Comment:
Yes it does, only with confirms I think you get the right number, or maybe using transactions which for this case is the same.